tests/commands/vmmap.py

```diff
@@ -48,3 +52,19 @@ def test_cmd_vmmap_name(self):
 
         res = gdb.execute("vmmap --name stack", to_string=True)
         self.assertEqual(len(res.splitlines()), 5)
+
+    def test_cmd_vmmap_perm(self):
+        gdb = self._gdb
+        gdb.execute("start")
+
+        res = gdb.execute("vmmap -p r?-", to_string=True)
+        self.assertGreater(len(res.splitlines()), 5)
+
+        res = gdb.execute("vmmap --perms r?-", to_string=True)
+        self.assertGreater(len(res.splitlines()), 5)
```

Comment:
You can go further in checking the output, this doesn't really check the output is as intended. Also those tests being equivalent, you can thoroughly check the strings in the 1st assert, then check that the 1st result equals the 2nd.

```suggestion
        res1 = gdb.execute("vmmap -p r?-", to_string=True)
        lines1 = res.splitlines()
        self.assertGreater(len(lines), 5)
        for line in lines1:
            perm_str = line.split()[3]
            assert perm_str[0] == 'r'
            assert perm_str[1] in ('w', '-')
            assert perm_str[2] == '-'

        res2 = gdb.execute("vmmap --perms r?-", to_string=True)
        assert res1 == res2
```